Quadriga/src/main/java/edu/asu/spring/quadriga/rest/SearchNetworkRestController.java

```
@@ -0,0 +1,128 @@
+    	
+    	List<IProject> projectList = new ArrayList<>();
+    	
+    	try {    		
+    	    for(String projectId : projectIds){
+    	    	IProject project = null;
+    	    	try{
+                    project = projectManager.getProjectDetails(projectId);
+    	    	}catch(QuadrigaStorageException e){
+    	    		logger.error("QuadrigaStorageException:", e);
+    	    	}
+                if(project!=null){
+                    projectList.add(project);
+                }
+            }
+    	    
+    		
+            if(projectList.size()==0){
+                String errorMsg = errorMessageRest.getErrorMsg("Projects don't exist.");
+                return new ResponseEntity<String>(errorMsg, HttpStatus.NOT_FOUND);
+            }
+
+            ITransformedNetwork transformedNetwork;
+            try{
+                transformedNetwork = transformationManager.getSearchTransformedNetworkMultipleProjects(
+        			projectIds, conceptId);	
+            }catch(QuadrigaStorageException e){
+                throw new RestException(403, e);
+	    	}
+            
+            
+    	    VelocityEngine engine = restVelocityFactory.getVelocityEngine(req);
+            engine.init();
+            Template template = engine.getTemplate("velocitytemplates/transformationdetails.vm");
+            VelocityContext context = new VelocityContext(restVelocityFactory.getVelocityContext());
+            context.put("nodeList", new ArrayList<Node>(transformedNetwork.getNodes().values()));
+            context.put("linkList", transformedNetwork.getLinks());
+            context.put("graphId", conceptId);
+            
+            StringWriter writer = new StringWriter();
+            template.merge(context, writer);
+            HttpHeaders httpHeaders = new HttpHeaders();
+            httpHeaders.setContentType(MediaType.APPLICATION_XML);
+            return new ResponseEntity<String>(writer.toString(), httpHeaders, HttpStatus.OK);
+        }catch (Exception e) {

```

Never have a catch all block like this. Catch the exceptions exactly where they are thrown. What are you catching here? Put it only around the code that throws it.